multiaccounts/accounts/database.go

```diff
@@ -49,44 +128,114 @@ func (db Database) Close() error {
 	return db.db.Close()
 }
 
-func (db *Database) SaveConfig(typ string, value interface{}) error {
-	_, err := db.db.Exec("INSERT OR REPLACE INTO settings (type, value) VALUES (?, ?)", typ, &sqlite.JSONBlob{value})
+func (db *Database) CreateSettings(s Settings, nodecfg params.NodeConfig) error {
+	_, err := db.db.Exec(`
+INSERT INTO settings (
+  address,
+  currency,
+  current_network,
+  dapps_address,
+  eip1581_address,
+  installation_id,
+  key_uid,
+  keycard_instance_uid,
+  keycard_paired_on,
+  keycard_pairing,
+  latest_derived_path,
+  mnemonic,
+  name,
+  networks,
+  node_config,
+  photo_path,
+  preview_privacy,
+  public_key,
+  signing_phrase,
+  wallet_root_address,
+  synthetic_id
+) VALUES (
+?,?,?,?,?,?,?,?,?,?,
+?,?,?,?,?,?,?,?,?,?,
+'id')`,
+		&sqlite.JSONBlob{s.Address},
+		&sqlite.JSONBlob{s.Currency},
+		&sqlite.JSONBlob{s.CurrentNetwork},
+		&sqlite.JSONBlob{s.DappsAddress},
+		&sqlite.JSONBlob{s.EIP1581Address},
+		&sqlite.JSONBlob{s.InstallationID},
+		&sqlite.JSONBlob{s.KeyUID},
+		&sqlite.JSONBlob{s.KeycardInstanceUID},
+		&sqlite.JSONBlob{s.KeycardPAiredOn},
+		&sqlite.JSONBlob{s.KeycardPairing},
+		&sqlite.JSONBlob{s.LatestDerivedPath},
+		&sqlite.JSONBlob{s.Mnemonic},
+		&sqlite.JSONBlob{s.Name},
+		&sqlite.JSONBlob{s.Networks},
+		&sqlite.JSONBlob{nodecfg},
+		&sqlite.JSONBlob{s.PhotoPath},
+		&sqlite.JSONBlob{s.PreviewPrivacy},
+		&sqlite.JSONBlob{s.PublicKey},
+		&sqlite.JSONBlob{s.SigningPhrase},
+		&sqlite.JSONBlob{s.WalletRootAddress})
+
 	return err
 }
 
-func (db *Database) GetConfig(typ string, value interface{}) error {
-	return db.db.QueryRow("SELECT value FROM settings WHERE type = ?", typ).Scan(&sqlite.JSONBlob{value})
+func (db *Database) SaveSetting(typ string, value interface{}) error {
+	setting, ok := settingsFields[typ]
+	if !ok {
+		return ErrInvalidConfig
+	}
+
+	stmt := fmt.Sprintf("UPDATE settings SET %s = ? WHERE synthetic_id = 'id'", setting)
+	_, err := db.db.Exec(stmt, &sqlite.JSONBlob{value})
```

Comment:
I think this statement forces us to use `JSONBlob` everywhere.
We should first ensure that things are stored correctly, i.e. If a save a string `hello` it should be stored as `hello`, not as `"hello"`, otherwise that might be annoying in the future.

So we should unwrap the type here of `interface{}` and have the right statement for each type, 
https://tour.golang.org/methods/16
that way we can ensure that things are stored in the right way and we can get rid of `JSONBlob{}`

Reply:
ah I see what we are trying to do,
I think we should just have a case on `typ` and parse the `interface{}` appropriately in the right type, so it gets stored correctly in the db, and we can get rid of the various `sqlite.JSONBlob`, might look boring but contains the complexity